web/src/elements/ak-mdx/ak-mdx.tsx

```diff
@@ -0,0 +1,172 @@
+};
+
+/**
+ * A replacer function that can be used to modify the output of the MDX component.
+ */
+export type Replacer = (input: string) => string;
+
+@customElement("ak-mdx")
+export class AKMDX extends AKElement {
+    @property({
+        reflect: true,
+    })
+    url: string = "";
+
+    @property()
+    content: string = "";
+
+    @property({ attribute: false })
+    replacers: Replacer[] = [];
+
+    #reactRoot: Root | null = null;
+
+    resolvedHTML = "";
+
+    static get styles(): CSSResult[] {
+        return [
+            PFList,
+            PFContent,
+            css`
+                h2:first-of-type {
+                    margin-top: 0;
+                }
+
+                svg[id^="mermaid-svg-"] {
+                    .rect {
+                        fill: var(
+                            --ak-mermaid-box-background-color,
+                            var(--pf-global--BackgroundColor--light-300)
+                        ) !important;
+                    }
+
+                    .messageText {
+                        stroke-width: 4;
+                        fill: var(--ak-mermaid-message-text) !important;
+                        paint-order: stroke;
+                    }
+                }
+            `,
+        ];
+    }
+
+    public async connectedCallback() {
+        super.connectedCallback();
+        this.#reactRoot = createRoot(this.shadowRoot!);
+
+        let nextMDXModule: MDXModule | undefined;
+
+        if (this.url) {
+            nextMDXModule = await fetchMDXModule(this.url);
+        } else {
+            nextMDXModule = {
+                content: this.content,
+            };
+        }
+
+        return this.delegateRender(nextMDXModule);
+    }
+
+    protected async delegateRender(mdxModule: MDXModule): Promise<void> {
+        if (!this.#reactRoot) return;
+
+        const normalized = this.replacers.reduce(
+            (content, replacer) => replacer(content),
+            mdxModule.content,
+        );
+
+        const mdx = await compileMDX(normalized, {
+            outputFormat: "function-body",
+            remarkPlugins: [
+                remarkParse,
+                remarkDirective,
+                remarkAdmonition,
+                remarkGFM,
+                remarkFrontmatter,
+                remarkMdxFrontmatter,
+                remarkHeadings,
+                remarkLists,
+            ],
+            rehypePlugins: [
+                // ---
+                [rehypeHighlight, highlightThemeOptions],
+                [
+                    rehypeMermaid,
+                    {
+                        prefix: "mermaid-svg-",
+                        colorScheme: this.activeTheme === UiThemeEnum.Dark ? "dark" : "light",
+                    } satisfies RehypeMermaidOptions,
+                ],
+            ],
+        });
+
+        const { default: Content, ...mdxExports } = await runMDX(mdx, {
+            ...runtime,
+            baseUrl: import.meta.url,
+        });
```

Comment:
Like our previous rendering with Showdown, this assumes trusted authorship of content. Blueprints may require another remark plugin to sanitize the HTML.